### PR TITLE
game-util/lutris: version bumps and dep fixups.

### PR DESCRIPTION
**Pull request by hanetzer**
Noticed my install of `game-util/lutris-9999` was not working all of a sudden. Apparently
they're moving into a new minor version which changed the deps. Also the current version
was missing an `xdg_desktop_database_update`.

### CODE REVIEW

**Comment by juippis** [comment]
Could you work together with #10533?
You both have 0.4.23 bump :)

**Comment by a17r** [comment]
Yes please, check if there's anything you have to add over 0.4.23 in the other PR. Also, both commits messages of yours would be missing actual version info. ;)

File: games-util/lutris/lutris-9999.ebuild [hunk 1]
```diff
@@ -27,9 +27,12 @@ RDEPEND="
 	dev-python/pygobject:3[${PYTHON_USEDEP}]
 	dev-python/python-evdev[${PYTHON_USEDEP}]
 	dev-python/pyyaml[${PYTHON_USEDEP}]
+	gnome-base/gnome-desktop
 	net-libs/libsoup
+	net-libs/webkit-gtk
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: net-libs/webkit-gtk:4[introspection]

File: games-util/lutris/lutris-9999.ebuild [hunk 2]
```diff
@@ -27,9 +27,12 @@ RDEPEND="
 	dev-python/pygobject:3[${PYTHON_USEDEP}]
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: You can get rid of dbus-python, it has been removed prior to 0.4.23

File: games-util/lutris/lutris-9999.ebuild [hunk 3]
```diff
@@ -27,9 +27,12 @@ RDEPEND="
...
 	dev-python/python-evdev[${PYTHON_USEDEP}]
 	dev-python/pyyaml[${PYTHON_USEDEP}]
+	gnome-base/gnome-desktop
 	net-libs/libsoup
+	net-libs/webkit-gtk
+	x11-apps/xgamma
 	x11-apps/xrandr
-	x11-apps/xgamma"
+"
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: I added additional deps specified by upstream in my version: app-arch/cabextract, edia-sound/fluid-soundfont, sys-process/psmisc. I wonder if some deps are needed at build time.

File: games-util/lutris/lutris-9999.ebuild [hunk 4]
```diff
@@ -27,9 +27,12 @@ RDEPEND="
...
 
 python_install() {
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: I'd do a
```
python_install_all() {
    local DOCS=( AUTHORS README.rst docs/installers.rst )
    distutils-r1_python_install_all
}
```

File: games-util/lutris/lutris-9999.ebuild [hunk 5]
```diff
@@ -49,14 +52,10 @@ src_install() {
 	distutils-r1_src_install
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: src_ functions can be removed, see my comment in python_install about $DOCS

File: games-util/lutris/lutris-9999.ebuild [hunk 6]
```diff
@@ -49,14 +52,10 @@ src_install() {
...
-	gnome2_icon_savelist
-	gnome2_schemas_savelist
-}
-
 pkg_postinst() {
 	gnome2_icon_cache_update
 	gnome2_schemas_update
+	xdg_desktop_database_update
 
 	elog "For a list of optional dependencies (runners) see:"
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: I could not find any "optional dep" in README.rst so I changed the text in my version:

```
# Quote README.rst
elog "Lutris installations are fully automated through scripts, which can"
elog "be written in either JSON or YAML. The scripting syntax is described"
elog "in ${EROOT%/}/usr/share/doc/${PF}/installers.rst.bz2, and is also"
elog "available online at lutris.net."
```

File: games-util/lutris/lutris-9999.ebuild [hunk 7]
```diff
@@ -49,14 +52,10 @@ src_install() {
 	distutils-r1_src_install
 }
```
Comment (nado) [inline games-util/lutris/lutris-9999.ebuild]: You can add tests, they work with dev-python/nose
```
python_test() {
    virtualx nosetests -v
}
```

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+else
+	SRC_URI="https://github.com/lutris/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+RDEPEND="
+	dev-python/dbus-python[${PYTHON_USEDEP}]
```
Comment (nado) [inline games-util/lutris/lutris-0.4.23.ebuild]: Can be removed

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+	dev-python/pygobject:3[${PYTHON_USEDEP}]
+	dev-python/python-evdev[${PYTHON_USEDEP}]
+	dev-python/pyyaml[${PYTHON_USEDEP}]
+	net-libs/libsoup
+	x11-apps/xrandr
+	x11-apps/xgamma"
```
Comment (nado) [inline games-util/lutris/lutris-0.4.23.ebuild]: app-arch/cabextract, app-arch/unrar, app-arch/unzip, app-arch/p7zip, x11-libs/gtk+:3[introspection] can be added.

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+python_install() {
```
Comment (nado) [inline games-util/lutris/lutris-0.4.23.ebuild]: For all functions, consider my comments for -9999 are the same here.

**Comment by nado** [comment]
Could you also mark your commits as closing bugs 667054 and 662374 (the latest is affected by reviews)?

**Comment by hanetzer** [comment]
@nado updating, have been busy.

File: games-util/lutris/lutris-9999.ebuild (hunk 7 above)
Comment (nado) [reply games-util/lutris/lutris-9999.ebuild]: Yup, they fire up a xvfb in their travis instance, so I just ran for virtualx.eclass, now that you mention it, I haven't run any test *without* virtualx.

**Comment by GuillaumeSeren** [comment]
Hey there,
I have been working on bumping this ebuild as well.
After discovering your work I close my PR to let you finish it and merge into portage.
I have been reviewing and testing your ebuild and the changes asked by @nado ,
this work well this is my upgraded version based on the work + feedback,
hope this can help !
```sh
# Copyright 1999-2019 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2
 
EAPI=6
       
PYTHON_COMPAT=( python3_{4,5,6} )
PYTHON_REQ_USE="sqlite,threads"

inherit distutils-r1 gnome2-utils xdg-utils

MY_PV="${PV/_beta/-beta}"
MY_P="${PN}-${MY_PV}"

DESCRIPTION="Lutris is an open source gaming platform for GNU/Linux."
HOMEPAGE="https://lutris.net/"

if [[ "${PV}" == "9999" ]] ; then
        EGIT_REPO_URI="https://github.com/lutris/lutris.git"
        inherit git-r3
else
        SRC_URI="https://github.com/lutris/${PN}/archive/v${MY_PV}.tar.gz -> ${MY_P}.tar.gz"
        KEYWORDS="~amd64 ~x86"
fi

LICENSE="GPL-3"
SLOT="0"

RDEPEND="
        app-arch/cabextract
        app-arch/p7zip
        app-arch/unrar
        app-arch/unzip
        dev-python/pillow[${PYTHON_USEDEP}]
        dev-python/pygobject:3[${PYTHON_USEDEP}]
        dev-python/python-evdev[${PYTHON_USEDEP}]
        dev-python/pyyaml[${PYTHON_USEDEP}]
        gnome-base/gnome-desktop[introspection]
        net-libs/libsoup
        net-libs/webkit-gtk:4[introspection]
        x11-apps/xgamma
        x11-apps/xrandr
        x11-libs/gtk+:3[introspection]
"

S="${WORKDIR}/${MY_P}"

python_install_all() {
    local DOCS=( AUTHORS README.rst docs/installers.rst )
    distutils-r1_python_install_all
}

pkg_postinst() {
        gnome2_icon_cache_update
        gnome2_schemas_update
        xdg_desktop_database_update

        # Quote README.rst
    elog "Lutris installations are fully automated through scripts, which can"
    elog "be written in either JSON or YAML. The scripting syntax is described"
    elog "in ${EROOT%/}/usr/share/doc/${PF}/installers.rst.bz2, and is also"
    elog "available online at lutris.net."
}

pkg_postrm() {
        gnome2_icon_cache_update
        gnome2_schemas_update
        xdg_desktop_database_update
}

```

**Comment by a17r** [comment]
The best way forward now would be to bump this to `0.5.0.1`.

**Comment by GuillaumeSeren** [comment]
@hanetzer version I upgraded is working for 0.5.0.1 on my station,
if I can provide more tests case please ask.

**Comment by Polynomial-C** [comment]
Alright guys, here's the deal. If you manage to get this PR in a shape where it is mergable within let's say a week from now on, I gonna do the merge. Otherwise I gonna try to get a good working ebuild into the tree all by myself. Sorry if that sounds harsh but all this mess with all these PRs that try to make lutris work again doesn't really lead anywhere. Upstream is already [not that amused](https://github.com/lutris/lutris/issues/1751) about what we're (not?) doing here.

Finally a more constructive note: I've [revbumped](https://github.com/gentoo/gentoo/commit/cb81ffd32cdda8ff5714c7a413a562a39421d0c1) `dev-python/python-evdev` so we can finally add python3_7 support to the lutris ebuild as well.

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 11]
```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+PYTHON_REQ_USE="sqlite,threads"
+
+inherit distutils-r1 gnome2-utils python-r1 xdg-utils
```
Comment (Polynomial-C) [inline games-util/lutris/lutris-0.4.23.ebuild]: Please replace `gnome2-utils` and `xdg-utils` with `xdg` eclass.

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+src_compile() {
+	distutils-r1_src_compile
+}
+
+src_install() {
+	# README.rst contains list of optional deps
+	DOCS=( AUTHORS README.rst INSTALL.rst )
+	distutils-r1_src_install
+}
+
```
Comment (Polynomial-C) [inline games-util/lutris/lutris-0.4.23.ebuild]: Please add

```
pkg_preinst() {
        xdg_pkg_preinst
}
```

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+pkg_postinst() {
+	gnome2_icon_cache_update
+	gnome2_schemas_update
+	xdg_desktop_database_update
```
Comment (Polynomial-C) [inline games-util/lutris/lutris-0.4.23.ebuild]: Please replace the above three lines with `xdg_pkg_postinst`

File: games-util/lutris/lutris-0.4.23.ebuild [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+	elog "For a list of optional dependencies (runners) see:"
+	elog "/usr/share/doc/${PF}/README.rst.bz2"
+}
+
+pkg_postrm() {
+	gnome2_icon_cache_update
+	gnome2_schemas_update
+	xdg_desktop_database_update
```
Comment (Polynomial-C) [inline games-util/lutris/lutris-0.4.23.ebuild]: Please replace the above three lines with `xdg_pkg_postrm`

File: games-util/lutris/lutris-0.4.23.ebuild (hunk 11 above)
Comment (juippis) [reply games-util/lutris/lutris-0.4.23.ebuild]: This should allow EAPI to be bumped to 7, too. Previous lutris versions worked without python-r1, so you can most likely drop that too, and delete those useless phases below.

**Comment by hanetzer** [comment]
@a17r @Polynomial-C should be in a usable state now. upstream apparently uses tar.xz
files hosted on lutris.net as official releases, but they do things which require S and P manipulation
(tarball named lutris_0.5.0.1.tar.xz, but it unpacks to a directory lutris [no version suffix]).
Should I do the manipulation or just continue using the tarballs from github which unpack
in the manner most easy to use with our build system (v0.5.0.1.tar.gz->lutris-0.5.0.1/)?

File: games-util/lutris/lutris-9999.ebuild (hunk 3 above)
Comment (a17r) [reply games-util/lutris/lutris-9999.ebuild]: What about `media-sound/fluid-soundfont`though?

File: games-util/lutris/lutris-9999.ebuild (hunk 1 above)
Comment (a17r) [reply games-util/lutris/lutris-9999.ebuild]: still appears to be missing slot (and probably slot operator as well?)

**Comment by a17r** [comment]
> files hosted on lutris.net as official releases, but they do things which require S and P manipulation
> (tarball named lutris_0.5.0.1.tar.xz, but it unpacks to a directory lutris [no version suffix]).
> Should I do the manipulation

Yes please, proper release tarballs are always preferred.

**Comment by a17r** [comment]
leaving final ack to @Polynomial-C

**Comment by Polynomial-C** [comment]
We still need `gnome-base/gnome-desktop[introspection]` in _RDEPEND_ or else lutris won't start.

File: games-util/lutris/lutris-0.5.0.1.ebuild [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+	xdg_pkg_preinst
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
+
+	# Quote README.rst
+	elog "Lutris installations are fully automated through scripts, which can"
+	elog "be written in either JSON or YAML. The scripting syntax is described"
+	elog "in ${EROOT%/}/usr/share/doc/${PF}/installers.rst.bz2, and is also"
```
Comment (Polynomial-C) [inline games-util/lutris/lutris-0.5.0.1.ebuild]: Starting with EAPI-7, trailing slashes are being stripped automatically. So please change `${EROOT%/}` to `${EROOT}`

**Pull request closed without merging**